### PR TITLE
Unit test for setLight function

### PR DESCRIPTION
**Pull request by Igor-Arena**

### CODE REVIEW

File: tests/unit_tests/setLight.spec.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { test, expect } from "@playwright/test"
+import { TrafficLight } from "../../pages/trafficLight"
+
+test("Testing setLight function", async () => {
+    const trafficLight = new TrafficLight;
+    await trafficLight.setLight(1);
+    expect(trafficLight.light).toBe("yellow");
+})
```
Comment (siarheima-papaya) [inline tests/unit_tests/setLight.spec.ts:8]: Correct. You are on the right way. 
Tip for this test: you can use parametrised test to check all lights

**Comment by Igor-Arena** [comment]
added next.spec.ts for testing next() function

**Comment by Igor-Arena** [comment]
1. Added tests for isGreenActive function
2. Added tests for isYellowActive function
3. Added tests for isRedActive function

File: tests/unit_tests/isGreenActive.spec.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { test, expect } from "@playwright/test";
```
Comment (siarheima-papaya) [inline tests/unit_tests/isGreenActive.spec.ts:1]: Please, don't split tests to different files. Create one file e.g. `trafficLightTest.spec.ts` and store all unit tests inside it

File: tests/unit_tests/isGreenActive.spec.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+import { TrafficLight } from "../../pages/trafficLight";
+
+test("Testing isGreenActive() function", async () => {
+  const trafficLight = new TrafficLight();
+  await trafficLight.setLight(0);
```
Comment (siarheima-papaya) [inline tests/unit_tests/isGreenActive.spec.ts:6]: Good job. Let's add negative scenario
Check how it works for index 1 and 2. You can use faker to generate random number from 1 to 2 and use it in test. Every test launch will create random value from 1 to 2 and allow checking different indexes every launch

Also, add negative scenario. What will happen if the index > 2?

File: tests/unit_tests/isRedActive.spec.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { test, expect } from "@playwright/test";
+import { TrafficLight } from "../../pages/trafficLight";
+
+test("Testing isRedActive() function", async () => {
```
Comment (siarheima-papaya) [inline tests/unit_tests/isRedActive.spec.ts:4]: The same as for `isGreenActive()`

File: tests/unit_tests/isYellowActive.spec.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { test, expect } from "@playwright/test";
+import { TrafficLight } from "../../pages/trafficLight";
+
+test("Testing isYellowActive() function", async () => {
```
Comment (siarheima-papaya) [inline tests/unit_tests/isYellowActive.spec.ts:4]: The same as for `isGreenActive()`

File: pages/trafficLight.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    static get colors(): string[] {
+        return ['green', 'yellow', 'red'];
+    }
+
+    get light(): string {
+        return TrafficLight.colors[this.lightIndex];
+    }
+
+    get colorsAmount(): number {
```
Comment (siarheima-papaya) [inline pages/trafficLight.ts:16]: Didn't covered by test

File: tests/unit_tests/next.spec.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    number: 2,
+    light: "red",
+    nextLight: "green",
+  },
+];
+
+for (const { number, light, nextLight } of lights) {
+  test(`Testing next() function with light ${light}`, async () => {
+    const trafficLight = new TrafficLight();
+    await trafficLight.setLight(`${number}`);
```
Comment (siarheima-papaya) [inline tests/unit_tests/next.spec.ts:25]: What will happen if the number > 2?

**Review by siarheima-papaya — APPROVED** [review]

**Pull request merged**